pkg/cdi/cache_test.go

```diff
@@ -812,6 +814,102 @@ devices:
 	}
 }
 
+func TestTooManyOpenFiles(t *testing.T) {
+	if runtime.GOOS != "linux" {
+		t.Skip("Test not applicable on Windows")
+	}
+
+	em, err := triggerEmfile()
+	require.NoError(t, err)
+	require.NotNil(t, em)
+	defer func() {
+		require.NoError(t, em.undo())
+	}()
+
+	_, err = syscall.Socket(syscall.AF_INET, syscall.SOCK_DGRAM, 0)
+	require.Equal(t, syscall.EMFILE, err)
+
+	cache := newCache(
+		WithAutoRefresh(true),
+	)
+	require.NotNil(t, cache)
+
+	// try to trigger original crash with a nil fsnotify.Watcher
+	_, _ = cache.InjectDevices(&oci.Spec{}, "vendor1.com/device=dev1")
```

Comment:
nit:
```suggestion
	devs := []string{"vendor1.com/device=dev1"}
	devices, err := cache.InjectDevices(&oci.Spec{}, devs...)
	require.Error(t, err)
	require.EqualValues(t, devs, devices)
```

Would it make sense to test other `cache` APIs as well?

Reply:
That InjectDevices should and will fail her (we have an empty cache, and we also have our file descriptors exhausted) but it should do it gracefully instead for crashing with a SIGSEGV (which it does until the next/tipmost commit is applied).

Reply:
Added a TestRecoveryAfterTooManyOpenFiles test case to verify that the cache becomes functional again if the process ever recovers from its inability to create file descriptors.